server/services/conditionsService.ts

```diff
@@ -187,41 +186,24 @@ function splitIntoGroupedObject(conditionObject, condition) {
 function populateFromSavedLicence(inputtedConditions: AdditionalConditions) {
   const populatedConditionIds = Object.keys(inputtedConditions)
 
-  return (condition) => {
-    const submission = getSubmissionForCondition(condition.id, inputtedConditions)
+  return (condition: ConditionMetadata) => {
+    const submission = getSubmissionForCondition(condition, inputtedConditions)
     const selected = populatedConditionIds.includes(String(condition.id))
 
     return { ...condition, selected, user_submission: submission }
   }
 }
 
-function getSubmissionForCondition(conditionId, inputtedConditions: AdditionalConditions) {
-  if (isEmpty(inputtedConditions[conditionId])) {
+function getSubmissionForCondition(condition: ConditionMetadata, inputtedConditions: AdditionalConditions) {
+  const { id } = condition
+  const inputtedCondition = inputtedConditions[id]
+  if (isEmpty(inputtedCondition)) {
     return {}
   }
 
-  if (conditionId === 'ATTENDDEPENDENCY') {
```

Comment:
pushed this logic into `displayForEdit` in the condition to make this function more generic and removed the requirement to have specific logic for individual conditions